specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+| `fadd`           | Used to add two floats together.  |                             |
+| `fsub`           | Used to subtract floats integers. |                             |
+| `fmul`           | Used to multiply floats           |                             |
+| `fdiv`           | Used for floating point division. | Can cause real-time errors. |
+|                  |                                   |                             |
+
+Finally, an adaptive profile program using fixed-point values (the `classical_fixed_points`
+module flag is set to `true`), can use the following intrinsics:
+
+| LLVM Intrinsic       | Context and Purpose                                  | Note                        |
+| :------------------- | :--------------------------------------------------- | :-------------------------- |
+| `llvm.smul.fix.*`    | Used to multiply two signed fixed point numbers.     |                             |
+| `llvm.smul.fix.sat*` | Same as above but clamps to min/max number in scale. |                             |
+| `llvm.umul.fix.*`    | Used to multiply two unsigned fixed point numbers.   |                             |
+| `llvm.umul.fix.sat*` | Same as above but clamps to min/max number in scale. |                             |
+| `llvm.sdiv.fix.*`    | Used to divide two signed fixed point numbers.       | Can cause real-time errors. |
+| `llvm.sdiv.fix.sat*` | Same as above but clamps to min/max number in scale. |                             |
+| `llvm.udiv.fix.*`    | Used to divide two unsigned fixed point numbers.     | Can cause real-time errors. |
+| `llvm.udiv.fix.sat*` | Same as above but clamps to min/max number in scale. |                             |
+|                      |                                                      |                             |
+
+### Bullet 10: User-defined functions and function calls
+
+An adaptive profile program may use user-defined functions and function calls.
+For example, consider that with user-defined functions if a backend has a `Cnot`
+gate in its instruction set, then a program that liberally uses `Swap` operations
+can define a function and call it as follows:
+
+```llvm
+define void @swap(%Qubit* %arg1, %Qubit* %arg2) {
+call void __quantum__qis__cnot__body(%Qubit* %arg1, %Qubit* %arg2)
+call void __quantum__qis__cnot__body(%Qubit* %arg2, %Qubit* %arg1)
+call void __quantum__qis__cnot__body(%Qubit* %arg1, %Qubit* %arg2)
+}
+
+define void @main() {
+...
+call void @swap_0_1(%Qubit* null, %Qubit* nonnull inttoptr (1 to %Qubit*))
+...
+}
+```
+
+Moreover, classical functions can be defined in the IR assuming that a backend
+has opted into supporting classical computations:
+
+```llvm
+define i64 @triple(i64 %0) {
+%1 = mul i64 %0, 3
+ret i64 %1
+}
+
+define void @main() {
+...
+%0 = call void @triple(i64 2)
+...
+}
+```
+
+The only restriction on user functions and definitions is that you cannot have
```

Comment:
I find this paragraph confusing. There are plenty of restrictions, such as e.g. what LLVM instructions can be used. Qubits are not special here - any restrictions that the profile imposes apply to any function. How about replacing this paragraph (except for the last sentence) with the sentence "All restrictions with respect to usage of data types and LLVM instructions defined by this profile also apply to user defined functions. In particular, the signatures of user-defined functions may only contain supported data types."